internal/operands/common-instancetypes/reconcile.go

```diff
@@ -58,52 +98,196 @@ func fetchClusterResources[C clusterType](bundlePath string) ([]C, error) {
 	}
 }
 
-func New(instancetypeBundlePath, preferenceBundlePath string) (operands.Operand, error) {
-	virtualMachineClusterInstancetypes, err := fetchClusterResources[instancetypev1alpha2.VirtualMachineClusterInstancetype](instancetypeBundlePath)
+func FetchBundleResource[C clusterType](path string) ([]C, error) {
+	file, err := ioutil.ReadFile(path)
 	if err != nil {
 		return nil, err
 	}
-	virtualMachineClusterPreferences, err := fetchClusterResources[instancetypev1alpha2.VirtualMachineClusterPreference](preferenceBundlePath)
+	return decodeResources[C](file)
+}
+
+func (c *commonInstancetypes) fetchResourcesFromBundle() ([]instancetypev1alpha2.VirtualMachineClusterInstancetype, []instancetypev1alpha2.VirtualMachineClusterPreference, error) {
+	virtualMachineClusterInstancetypes, err := FetchBundleResource[instancetypev1alpha2.VirtualMachineClusterInstancetype](c.virtualMachineClusterInstancetypeBundle)
+	if err != nil {
+		return nil, nil, err
+	}
+	virtualMachineClusterPreferences, err := FetchBundleResource[instancetypev1alpha2.VirtualMachineClusterPreference](c.virtualMachineClusterPreferenceBundle)
+	if err != nil {
+		return nil, nil, err
+	}
+	return virtualMachineClusterInstancetypes, virtualMachineClusterPreferences, err
+}
+
+func generateKustomization(URL string) (filesys.FileSystem, string, error) {
+	fSys := filesys.MakeFsOnDisk()
+	tmpDir, err := filesys.NewTmpConfirmedDir()
```

Comment:
I think the temporary directory should be removed when it's no longer needed, otherwise it is a leak.